### PR TITLE
Run mocha test in CI using selenium

### PR DESCRIPTION
**Pull request by bgrins**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI
+
+on:
+  push:
+    branches: [ "main", "tests-in-ci" ]
```
Comment (bgrins) [inline .github/workflows/test.yml:5]: Would remove this branch before committing - this is allowing results to show up in https://github.com/bgrins/Speedometer/actions

**Comment by bgrins** [comment]
@housseindjirdeh a little rough but this is what I was thinking. Open to suggestion on making the actual runner better but it seems to work alright.

**Comment by bgrins** [comment]
This can be run locally with for example `BROWSER=firefox node tests/unit-test-run.mjs`. Browser specific notes:
- Firefox requires geckodriver https://github.com/mozilla/geckodriver/releases. On mac you can install with `brew install geckodriver` as well.
- Safari requires you to enable safaridriver via https://developer.apple.com/documentation/webkit/testing_with_webdriver_in_safari.
- I think it finds Chrome without any additional install - would be curious if it works for you locally.

File: tests/unit-test-run.mjs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+async function test() {
+  let capabilities;
+  switch (process.env.BROWSER) {
+    case "safari":
+      capabilities = Capabilities.safari();
+      capabilities.set("safari.options", { technologyPreview: false });
+      break;
+
+    case "firefox": {
+      capabilities = Capabilities.firefox().setLoggingPrefs({ browser: "ALL" });
```
Comment (rniwa) [inline tests/unit-test-run.mjs:24]: Doesn't enabling extra logging  affect perf?

File: tests/unit-test-run.mjs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+
+import serve from "./unit-test-server.mjs";
+import {
+  Builder,
```
Comment (rniwa) [inline tests/unit-test-run.mjs:4]: Please use 4-space indentation per coding style guideline.

File: tests/unit-test-run.mjs (hunk 2 above)
Comment (bgrins) [reply tests/unit-test-run.mjs:24]: Maybe, but this isn't actually running the benchmark or reporting results, this is only running the unit tests

File: tests/unit-test-run.mjs (hunk 3 above)
Comment (bgrins) [reply tests/unit-test-run.mjs:4]: OK. Looking forward to auto formatting

File: tests/unit-test-run.mjs (hunk 2 above)
Comment (bgrins) [reply tests/unit-test-run.mjs]: I'll just remove that from all of the browsers so it's consistent though, as I'm not seeing an observable change with or without it

File: tests/unit-test-run.mjs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import serve from "./unit-test-server.mjs";
+import { Builder, By, Capabilities } from "selenium-webdriver";
```
Comment (housseindjirdeh) [inline tests/unit-test-run.mjs:2]:
```suggestion
import { Builder, Capabilities } from "selenium-webdriver";
```

File: tests/unit-test-server.mjs [hunk 5]
```diff
@@ -0,0 +1,50 @@
+// Simple server adapted from https://developer.mozilla.org/en-US/docs/Learn/Server-side/Node_server_without_framework:
```
Comment (housseindjirdeh) [inline tests/unit-test-server.mjs:1]: Nit: I assume we'll likely continue to use our testing setup even when we introduce larger end-to-end tests? If that's the case, maybe we can just rename these files to `run.mjs` and `server.mjs`?

**Review by housseindjirdeh — APPROVED** [review]
Just minor nits. Otherwise this looks great 🥳 

> I think it finds Chrome without any additional install - would be curious if it works for you locally.

It does, I just had to update my outdated chromedriver and it worked (`brew upgrade chromedriver`)

File: tests/unit-test-server.mjs (hunk 5 above)
Comment (bgrins) [reply tests/unit-test-server.mjs:1]: Yeah, we may want to pull in some dependencies if this gets more complex but the generic names make sense.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+name: CI
+
+on:
+  push:
```
Comment (rniwa) [inline .github/workflows/test.yml:4]: Use 4 space indentation?

**Review by rniwa — APPROVED** [review]

File: .github/workflows/test.yml (hunk 6 above)
Comment (bgrins) [reply .github/workflows/test.yml:4]: I just copied this from github actions template, I think 2 space is conventional here

**Pull request merged**